pkg/types/types.go

```diff
@@ -4,12 +4,89 @@
 package types
 
 import (
+	"path"
+	"strings"
+
+	"github.com/square/p2/pkg/util"
+
+	"github.com/pborman/uuid"
 	"k8s.io/kubernetes/pkg/util/sets"
 )
 
 type NodeName string
+
+// Refers to the id: key in a pod manifest, i.e. the name of the application
+// running in the pod.  There may be multiple copies (pods) of a given pod id
+// running at a given time
 type PodID string
 
+// A unique identifier for each pod (instance). At the time of defining this
+// type a PodUniqueKey is is the hostname of the node on which the pod is running
+// and the pod id joined with a slash, e.g. "example.com/mysql" if a pod with id
+// "mysql" is running on a node "example.com".
+//
+// P2 will begin using uuids instead of the previous format to better support
+// running multiple pods with the same pod id on the same node. Certain new
+// functionality will only be supported for pods with UUID unique keys, hence
+// the need for the bool.
+type PodUniqueKey struct {
+	// The actual key, e.g. "example.com/mysql" under the old format or a uuid
+	// under the new format
+	id string
+
+	// Tracks whether the key is a uuid or the node/pod_id concatenation. Some
+	// features will require isUUID to be true.
+	isUUID bool
```

Comment:
alternatively I could make `PodUniqueKey` _always_ be a uuid, and then in `ManifestResult` it would be a pointer that is nil if the pod is using the old key format.